### DISCUSSION TITLE
Help with installation in restricted school Windows environments

### DISCUSSION THREAD

**Discussion by villares** [jpype-project/jpype#1094] [Q&A] [answered]
I'm trying to install **jpype** on some Windows machines where it is hard to obtain much permissions (bureocratic school env). I'm using a portable Thonny IDE (that has an embeded Python 3.8 or 3.10) and I tried installing on a different machine and moving but it didn't work. 

When I try installign it with pip I see the C++ build tools are missing, but I've been unable to install them:
```
error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
``` 
Any suggestions on how to overcome this?

> Context is I want to use https://pycoding.org, that depends on **jpype**, with students https://twitter.com/villares/status/1541096458304339970?s=20&t=ugLPqoZn85sRlD67wCcnYQ

**Comment by Thrameos** [comment]
Is there some reason the binary wheel didn't work?    Building the wheel and installing would be my first approach.    Even directly copying the jpype site directory with the corresponding jar file should work.

**Reply by villares** [reply]
Thanks for the interest @Thrameos, I have no idea why it didn't work, and unfortunately I have very little knowledge of packaging/wheels. If you could give me some instructions on how to investigate this, with some patience (as I have access to those computers only twice a week), I could report back. 

The portable proof of concept is here: 
https://twitter.com/villares/status/1541096458304339970?t=_YcS9Osf46ew64gitlFW2Q&s=19

I thought it should work but it didn't.

**Reply by villares** [reply]
To clarify, I don't know anything about building wheels. From a quick search I understand I should clone the repo and try `python setup.py bdist_wheel`. Do I need to do it in a similar Windows machine to the one I wish to install to?

**Reply by Thrameos** [reply]
The patterns for building are all in the .azure directory.   That is how we produce all the distributions.   You will need a machine with the same Python and similar libraries in terms of Visual Studio and Java.   Assuming you have Python and the visual studio runtime installed on the target machine it should just work.

**Reply by villares** [reply]
Ok, this might take some time. I have no Visual Studio. I have a JDK-18 I use with jpype. I'll see if can get some time to try it and then I'll reach back here if I stumble. Thank you!

**Comment by pelson** [comment]
My personal opinion: I would reach for ``conda``. It is an easy way to install Python, complex Python packages (JPype) and even OpenJDK all in a single/consistent way on basically any platform. Unfortunately it isn't ``pip`` though...

**Reply by villares** [reply]
I've been unable to investigate further this issue. Thanks for the suggestion @pelson! I think that if I can install `conda` on that problematic machine it might be the easiest way to achieve the original aim of getting jpype working. If I can do it I'll report back here.

**Comment by villares** [comment, marked as answer]
I'm going to "close" this, as I don't have access to those computers anymore.
Thank you for the caring interest and support @Thrameos and @pelson!
